src/components/ZoomableImage.vue

```diff
@@ -0,0 +1,90 @@
+<script setup>
+import { defineProps, ref} from 'vue'
+// eslint-disable-next-line no-unused-vars
+const props = defineProps(['imageSrc'])
+const image = ref(null)
+const imageLeftPos = ref(0)
+const imageTopPos = ref(0)
+const currentZoom = ref(1)
+
+const MIN_ZOOM = 1
+const MAX_ZOOM = 3
+const STEP_SIZE = 0.1
+let mouseX
+let mouseY
+
```

Comment:
Zoom will be run whenever the magnify buttons are pressed, mouse wheel is scrolled, or fingers pinched. It takes an optional arg which is direction and then uses the stepsize to change the images scale accordingly. Overflow on the image's container is hidden which gives it a zoom effect